README.md

```diff
@@ -1,10 +1,376 @@
+
+```toml
+# wrangler.toml
+
+name = "api"
+main = "./api/index.ts"
+compatibility_date = "2024-12-05"
+assets = { not_found_handling = "single-page-application", binding = "ASSETS" }
+```
+
+The assets `binding` defined here will allow us to access the assets functionality from our Worker.
+
+### Add your API Worker
+
+```ts
+// api/index.ts
+
+interface Env {
+  ASSETS: Fetcher;
+}
+
+export default {
+  fetch(request, env) {
+    const url = new URL(request.url);
+
+    if (url.pathname.startsWith('/api/')) {
+      return Response.json({
+        name: 'Cloudflare',
+      });
+    }
+
+    return env.ASSETS.fetch(request);
+  },
+} satisfies ExportedHandler<Env>;
+```
+
+The Worker above will be invoked for any not found path.
+It returns a JSON response if the `pathname` starts with `/api/` and otherwise passes the incoming request through to the asset binding.
+This means that for paths that do not start with `/api/`, the `not_found_handling` behaviour defined in the Worker config will be evaluated and the `index.html` file will be returned.
+
+### Call the API from the client
+
+Edit `src/App.tsx` so that it includes an additional button that calls the API and sets some state. You can replace the file contents with the following code.
+
+```tsx
+// src/App.tsx
+
+import { useState } from 'react';
+import reactLogo from './assets/react.svg';
+import viteLogo from '/vite.svg';
+import './App.css';
+
+function App() {
+  const [count, setCount] = useState(0);
+  const [name, setName] = useState('unknown');
+
+  return (
+    <>
+      <div>
+        <a href="https://vite.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>Vite + React</h1>
+      <div className="card">
+        <button
+          onClick={() => setCount((count) => count + 1)}
+          aria-label="increment"
+        >
+          count is {count}
+        </button>
+        <p>
+          Edit <code>src/App.tsx</code> and save to test HMR
+        </p>
+      </div>
+      <div className="card">
+        <button
+          onClick={() => {
+            fetch('/api/')
+              .then((res) => res.json() as Promise<{ name: string }>)
+              .then((data) => setName(data.name));
+          }}
+          aria-label="get-name"
+        >
+          Name from API is: {name}
+        </button>
+        <p>
+          Edit <code>api/index.ts</code> to change the name
+        </p>
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+    </>
+  );
+}
+
+export default App;
 ```
+
+Now, if you click the button, it will display 'Name from API is: Cloudflare'.
+
+Try incrementing the counter and then changing the `name` that is returned in `api/index.ts`.
```

Comment:
consider being a bit more explicit here, as the instructions are a bit vague and the impact might not be understood, for example (feel free to rephrase):

> Let's now see hot module reloading in action. Try incrementing the counter to initialize application state in the browser. Then edit `api/index.ts` by changing the `name` it returns to `"Cloudflare Workers"`. If you now click the button again, it will display the new `name` while preserving the previously set counter value! You can now iterate on your app quickly without losing any of the UI state in between edits.